locks.py

```diff
@@ -1,47 +1,25 @@
-from twisted.internet import defer
+import os
 
-from buildbot.plugins import *
-from buildbot.process.properties import Properties, Property
-from buildbot.process.remotecommand import RemoteCommand
-from buildbot.steps.mtrlogobserver import MTR, MtrLogObserver
-from buildbot.steps.shell import Compile, SetPropertyFromCommand, ShellCommand, Test
-from buildbot.steps.source.github import GitHub
-from constants import *
+import yaml
 
-####### LOCKS
-# main_master_lock = util.MasterLock('main_master_lock', maxCount=30)
+from buildbot.plugins import util
 
-hz_bbw1_lock = util.MasterLock("hz_bbw1_lock", maxCount=9)
-hz_bbw2_lock = util.MasterLock("hz_bbw2_lock", maxCount=1)
-hz_bbw4_lock = util.MasterLock("hz_bbw4_lock", maxCount=9)
-hz_bbw5_lock = util.MasterLock("hz_bbw5_lock", maxCount=9)
-amd_bbw1_lock = util.MasterLock("amd_bbw1_lock", maxCount=4)
-amd_bbw2_lock = util.MasterLock("amd_bbw2_lock", maxCount=6)
-intel_bbw1_lock = util.MasterLock("intel_bbw1_lock", maxCount=5)
-p9_rhel8_bbw1_lock = util.MasterLock("p9_rhel8_bbw1_lock", maxCount=6)
-p9_db_bbw1_lock = util.MasterLock("p9_db_bbw1_lock", maxCount=8)
-p9_raptor_bbw1_lock = util.MasterLock("p9_raptor_bbw1_lock", maxCount=6)
-aarch_bbw1_lock = util.MasterLock("aarch64_bbw1_lock", maxCount=2)
-aarch_bbw2_lock = util.MasterLock("aarch64_bbw2_lock", maxCount=2)
-aarch_bbw3_lock = util.MasterLock("aarch64_bbw3_lock", maxCount=2)
-aarch_bbw4_lock = util.MasterLock("aarch64_bbw4_lock", maxCount=2)
-aarch_bbw5_lock = util.MasterLock("aarch64_bbw5_lock", maxCount=15)
-aarch_bbw6_lock = util.MasterLock("aarch64_bbw6_lock", maxCount=15)
-aarch_bbw7_lock = util.MasterLock("aarch64_bbw7_lock", maxCount=15)
-apexis_bbw1_lock = util.MasterLock("apexis_bbw1_lock", maxCount=1)
-apexis_bbw2_lock = util.MasterLock("apexis_bbw2_lock", maxCount=1)
-apexis_bbw3_lock = util.MasterLock("apexis_bbw3_lock", maxCount=6)
-bg_bbw1_lock = util.MasterLock("bg_bbw1_lock", maxCount=3)
-bg_bbw2_lock = util.MasterLock("bg_bbw2_lock", maxCount=2)
-bg_bbw3_lock = util.MasterLock("bg_bbw3_lock", maxCount=2)
-bg_bbw4_lock = util.MasterLock("bg_bbw4_lock", maxCount=2)
-win_bbw1_lock = util.MasterLock("win_bbw1_lock", maxCount=1)
-win_bbw2_lock = util.MasterLock("win_bbw2_lock", maxCount=4)
-s390x_bbw1_lock = util.MasterLock("s390x_bbw1_lock", maxCount=3)
-s390x_bbw2_lock = util.MasterLock("s390x_bbw2_lock", maxCount=3)
-s390x_bbw3_lock = util.MasterLock("s390x_bbw3_lock", maxCount=3)
-s390x_bbw4_lock = util.MasterLock("s390x_bbw4_lock", maxCount=3)
-s390x_bbw5_lock = util.MasterLock("s390x_bbw5_lock", maxCount=3)
+# Local
+from constants import builders_install, builders_upgrade, github_status_builders
+
+LOCKS: dict[str, util.MasterLock] = {}
+# worker_locks.yaml currently is in the same folder as locks.py.
+# TODO: re-evaluate if this is the right place after multi-master
+# is refactored to use a single base master.cfg.
+with open(
+    os.path.join(os.path.dirname(os.path.realpath(__file__)), "worker_locks.yaml"),
```

Comment:
@cvicentiu Why not keep the locks in constants, like every constant?
So we avoid any additional files that hold constant values.

```
WORKER_LOCKS = {
    "hz-bbw1-docker": 9,
    "hz-bbw2-docker": 1,
}
```

Reply:
We talked over the phone and agreed that this configuration data should stay together with other worker metadata.